src/highlight/types.ts

```diff
@@ -1,18 +1,24 @@
-export interface PostJointRequest {
-  unit: PendingUnit;
-}
+import { TypedDataDomain } from '@ethersproject/abstract-signer';
+
+export type PostMessageRequest = Message;
+
+export type Domain = Required<TypedDataDomain>;
+
+export type Message = {
+  domain: Domain;
+  message: Record<string, unknown>;
+  entrypoint: string;
```

Comment:
Is the entrypoint signed by the user? If that's not the case and there is an agent with 2 function having same params like `add(a, b, c)` and `delete(a, b, c)`, someone could trigger `delete` instead of `add`.

Reply:
It’s not signed. To make it replayable for different entry point agent would need to have the same type hash as well.

Reply:
To make it failproof if agent creates multiple endpoints with same struct hash we would just need to put it inside message itself. Every agent would need to define their messages with entry point in mind.